datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt

```diff
@@ -177,12 +195,38 @@ private fun Questionnaire.QuestionnaireItemComponent.createQuestionnaireResponse
   return QuestionnaireResponse.QuestionnaireResponseItemComponent().apply {
     linkId = this@createQuestionnaireResponseItem.linkId
     answer = createQuestionnaireResponseItemAnswers()
-    this@createQuestionnaireResponseItem.item.forEach {
-      this.addItem(it.createQuestionnaireResponseItem())
+    if (this@createQuestionnaireResponseItem.type != Questionnaire.QuestionnaireItemType.GROUP &&
+        this@createQuestionnaireResponseItem.item.count() > 0
+    ) {
+      if (hasNestedItemsWithinAnswers && answer.isNotEmpty()) {
+        this.addNestedItemsToAnswer(this@createQuestionnaireResponseItem)
+      }
+    } else if (this@createQuestionnaireResponseItem.type ==
+        Questionnaire.QuestionnaireItemType.GROUP
+    ) {
+      this@createQuestionnaireResponseItem.item.forEach {
+        this.addItem(it.createQuestionnaireResponseItem())
+      }
     }
```

Comment:
What I mean here is this:

```suggestion
    if (answer.isNotEmpty() && hasNestedItemsWithinAnswers) {
        this.addNestedItemsToAnswer(this@createQuestionnaireResponseItem)
    }
    if (hasNestedItemsWithinItems) {
      this@createQuestionnaireResponseItem.item.forEach {
        this.addItem(it.createQuestionnaireResponseItem())
      }
    }
```

and then add 

```
val Questionnaire.QuestionnaireItemComponent.hasNestedItemsWithinItems: Boolean
  get() = item.isNotEmpty() && type == Questionnaire.QuestionnaireItemType.GROUP
```

in `MoreQuestionnaireItemOptions.kt`

I think the conditional block seems more logical. What do you think?

Reply:
oh thats weird.. i think you reviewed an older version of the file if you check the [branch](https://github.com/google/android-fhir/tree/kj/sdc-nested-item-in-answer), you will see that this is already done..